eventstore.go

```diff
@@ -152,24 +155,14 @@ func UpToSequence(seq uint64) EvolveOption {
 	})
 }
 
-type natsApiError struct {
-	Code        int    `json:"code"`
-	ErrCode     uint16 `json:"err_code"`
-	Description string `json:"description"`
-}
-
-type natsGetMsgRequest struct {
-	LastBySubject string `json:"last_by_subj"`
-}
-
-type natsGetMsgResponse struct {
-	Type    string         `json:"type"`
-	Error   *natsApiError  `json:"error"`
-	Message *natsStoredMsg `json:"message"`
-}
-
-type natsStoredMsg struct {
-	Sequence uint64 `json:"seq"`
+// Filters specifies the subject filter to use when evolving state.
+// The filter can be in the form of `<entity-type>`, `<entity-type>.<entity-id>`,
+// or `<entity-type>.<entity-id>.<event-type>`. Wildcards can be used as well.
+func Filters(filters ...string) EvolveOption {
```

Comment:
@jordan-rash I chose to go this route as an option to pass specific filters to `Evolve` instead of making it an optional interface to satisfy on the model. It provides more flexibility in that you can use the same model, but with a different subset of filters to derive state for different things, e.g. `en.1` and `en.2` without needing declare in the model ahead of time.